rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -22,10 +22,9 @@ nested_array_dict = {}
 @{have_not_included_string = False}@
 #include <rosidl_generator_c/string.h>
 #include <rosidl_generator_c/string_functions.h>
-
 @[  end if]@
 @{
-if not field.type.is_primitive_type() and field.type.is_array:
+if not field.type.is_primitive_type():
```

Comment:
We don't need to relax it anymore, I think. The original fix needed a bunch of array destroy functions and whatnot.